### ISSUE TITLE
Example run-transactions-complete, not printing storage content

### ISSUE THREAD

**Issue by blintgit** [ethereumjs/ethereumjs-vm#234]
Running the example mentioned in the title, there is no output after the "---Storage---" line. According to my understanding, there should be a null_radix stored at the created address. (running gives no exceptions)
Please correct me if I'm wrong! Thanks

**Comment by blintgit** [comment]
[update]

At the following part of the example at the `runTx` function I've noticed that we refresh the `createdAddress` variable at every transaction even though it doesn't register a new contract. Therefore it leaves us with the loss of our contract's address and the `readStorage` function won't print any output.

I've fixed it by checking the `createdAddress` property for `null` before saving the value.

```
if (returns.createdAddress) {
  createdAddress = results.createdAddress
  console.log('address created: ' +
      createdAddress.toString('hex'))
}
```

**Comment by holgerd77** [comment]
Hi @blintgit, maybe wait up a couple of more contributions before talking in the "we" form, this is a bit irritating. :-)

That said: good catch, can you prepare a PR with your proposed changes?

**Comment by blintgit** [comment]
Yes, I'm preparing the PR asap. Thanks

**Comment by hswick** [comment]
returns is not a defined variable, believe you meant results. However, this doesn't seem to fix the issue for me.

**Comment by rosulucian** [comment]
same as @hswick : 
i added 
`    if (!createdAddress) {
      createdAddress = results.createdAddress
    }`

so the address would not reset but I set get an empty trie and no storage is printed

**Comment by rosulucian** [comment]
I had another look at it and after calling runCode I tried getting the account so I can read it`s trie by calling either stateTrie.get(createdAddress) or res.runState.stateManager.getAccount(res.runState.address) but I only get an empty account, i.e. codehash is SHA3_NULL_S and stateRoot is SHA3_RLP_S.

The I tried to get the storage trie through res.runState.stateManager._getStorageTrie(res.runState.address) and for some reason you still find it in the state manager it just didn`t commit somehow. on the other hand the trie i got through stateManager._getStorageTrie() has the keys off, e.g. when deploying a contract I got the member variable values right but the keys are not as per assembly, i.e. starting with 0x0.
